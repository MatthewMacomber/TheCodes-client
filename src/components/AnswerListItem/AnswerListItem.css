.AnswerListItem {
  position: relative;
  width: 100%;
  max-width: 300px;
  height: 100%;
  max-height: 200px;
}

.image {
  display: block;
  width: 100%;
  height: auto;
}

.overlay {
  position: absolute;
  bottom: 0;
  background: rgb(0,0,0);
  color: #f1f1f1;
  width: 100%;
  font-size: 15px;
  padding: 0;
  margin: 0;
  text-align: center;
}

.overlay > h3 {
  margin: 0;
}