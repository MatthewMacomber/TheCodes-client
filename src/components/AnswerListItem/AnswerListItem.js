import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import {NiceDate, Hyph} from '../Utils/Utils';

export default class AnswerListItem extends Component {
  render() {
    const {answer} = this.props;
    return  (
      <Link to={`/answer/${answer.id}`} className='AnswerListItem'>
        <header className='AnswerListItem__header'>
          <h2 className='AnswerListItem__header'>
            Answer for code id: {answer.code_id}
          </h2>
          <NiceDate date={answer.date_created} />
        </header>
        <h3>{answer.correct ? <>Correct</> : <>Wrong</>}</h3>
        <footer>
          Answer ID <Hyph /> {answer.id}
        </footer>
      </Link>
    )
  }
}