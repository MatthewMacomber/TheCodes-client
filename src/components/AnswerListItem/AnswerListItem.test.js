import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import AnswerListItem from './AnswerListItem';

Enzyme.configure({ adapter: new Adapter() })

describe('CodeListItem component', () => {
  const props = {
    id: 1,
    title: 'code',
    content: 'test code',
    user_name: 'test',
    date_created: new Date(2018, 12, 15)
  }

  it('renders the AnswerListItem given props', () => {
    const wrapper = shallow(<AnswerListItem answer={props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});