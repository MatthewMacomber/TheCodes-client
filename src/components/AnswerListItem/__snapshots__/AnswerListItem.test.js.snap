// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeListItem component renders the AnswerListItem given props 1`] = `
<Link
  className="AnswerListItem"
  to="/answer/1"
>
  <img
    alt="test code"
    class="image"
    src=""
  />
  <div
    className="overlay"
  >
    <header
      className="AnswerListItem__header"
    >
      <h2
        className="AnswerListItem__header"
      >
        Answer for code id: 
      </h2>
      <NiceDate
        date={2019-01-15T08:00:00.000Z}
      />
    </header>
    <h3>
      Wrong
    </h3>
    <footer>
      Answer ID 
      <Hyph />
       
      1
    </footer>
  </div>
</Link>
`;
